### DISCUSSION TITLE
Element picker highlights more than it should

### DISCUSSION THREAD

**Discussion by MasterKia** [uBlockOrigin/uBlock-issues#2538] [Q&A] [answered]
Go to `tarfandha.org`; open the element picker, enter:
```adb
##a[title][rel][alt]:upward(1)
```
OR
```adb
##[style="border: 1px solid #000;text-align: center;text-decoration: none !important;"]:matches-css(padding-right: /^/)
```
OR
```adb
##.behtarinseo
```

Notice that uBO highlights the whole sidebar.
![Screenshot_20230311_225025](https://user-images.githubusercontent.com/17685483/224507612-e9f97a5d-19fc-4f84-8452-21d641ba937f.png)

Press "Preview", notice that only links below the "پرسش و پاسخ" header will get hidden _and not the whole sidebar._
![Screenshot_20230311_225041](https://user-images.githubusercontent.com/17685483/224507625-3482c4dc-b4e3-4a7f-a3d6-e0467dc7d90c.png)

1.47.5b6 dev, Firefox 110

**Comment by gorhill** [comment, marked as answer]
I get 22 matches with `document.querySelectorAll('a[title][rel][alt]');` from console.

Why do you think uBO highlights more than it should?

---

The first `a` returned in the set above has a parent that matches what the picker highlights. Just use the browser dev tools when in doubt, they should clear up any assumptions.

**Reply by MasterKia** [reply]
In this picture when you press "Preview" you'd expect _the whole highlighted area_ to be hidden:
https://user-images.githubusercontent.com/17685483/224507612-e9f97a5d-19fc-4f84-8452-21d641ba937f.png

But after you press "Preview" only a certain part gets hidden (the part under "پرسش و پاسخ"):
https://user-images.githubusercontent.com/17685483/224507625-3482c4dc-b4e3-4a7f-a3d6-e0467dc7d90c.png

**Reply by gorhill** [reply]
Use browser dev tools, you will understand uBO does exactly what your filter asks.

---

All the elements which come before `.behtarinseo` have `float: right` on them, this is the result -- uBO is just showing you exactly what the browser inspector shows you when you selected the same element.

**Reply by MasterKia** [reply]
My filter only wants to hide the part I specified in blue, but uBO highlights the whole sidebar as if my filter wants to hide the sidebar:

https://user-images.githubusercontent.com/17685483/224508130-1f9d5d41-fb2e-4132-bc37-8d7a9377475a.png

In the end, my filter only hides the blue part and not the whole sidebar which uBO had highlighted.

**Reply by MasterKia** [reply]
The "Preview" button of uBO does apply the filter correctly as I wanted it to, but it just highlights the wrong parts; that's all.

**Reply by gorhill** [reply]
The browser dev tools also highlight the "wrong" part -- why would you think uBO can do better? The part is not "wrong", it's literally the dimension of the selected element, as returned by the browser API. The whole thing is the result of the page using `float: right` on elements in the sidebar. The browser dev tools is the truth about what is going on, uBO matches exactly what these tools say.

**Reply by MasterKia** [reply]
True, if I remove `float:right` from the elements above `##.behatrinseo` then this "behavior" does not occur anymore.

> _it's literally the dimension of the selected element_, as returned by the browser API. The whole thing is the result of the page using `float: right` on elements in the sidebar.

Don't you find this behavior of the browser API unexpected?

**Reply by gorhill** [reply]
Not really, that expected with `float: right`, the flow of inline elements has to be rendered in such way as to avoid being rendered under the floating elements, and in this case they end up being wholly pushed down because there is no place horizontally to render them beside the floating elements.

**Comment by MasterKia** [comment]
Hmm, I checked the dev tools. Very strange.

![Screenshot_20230311_230929](https://user-images.githubusercontent.com/17685483/224508334-74b7481d-bf4f-45c9-af12-96ebb6b09b81.png)


![Screenshot_20230311_231004](https://user-images.githubusercontent.com/17685483/224508348-7d7d684f-bc70-4524-a484-37d06dc89dd1.png)

**Comment by krystian3w** [comment]
AdGuard maybe implement in future: https://github.com/AdguardTeam/ExtendedCss/issues/124
